go/vt/sqlparser/parsed_query.go

```diff
@@ -44,6 +45,16 @@ func NewParsedQuery(node SQLNode) *ParsedQuery {
 	return buf.ParsedQuery()
 }
 
+// CachedSize returns the approximate size that this query takes when cached in memory
+func (pq *ParsedQuery) CachedSize() int64 {
+	if pq == nil {
+		return 0
+	}
+	return int64(unsafe.Sizeof(ParsedQuery{})) +
```

Comment:
historically, we have relegated the use of unsafe to https://github.com/vitessio/vitess/blob/master/go/hack/hack.go. Is it time to let go of that rule, @sougou? (I think it is)